### PR TITLE
Hide Password mismatch incase both the passwords are empty

### PR DESCRIPTION
**Pull request by mananjadhav**
@alex-mechler Can you please help review this?

### Details
Show Password mismatch error only if New Password is entered.

Regression from #4367 

### Fixed Issues

$ #4606 
$ https://github.com/Expensify/App/issues/4665

### Tests
Apart from the test cases mentioned here #4409 
1. Tested by keeping New Password and Confirm Password blank onBlur.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/3069065/129234939-5848e435-5263-4524-91f6-170f3fc7810c.mp4

#### Mobile Web

https://user-images.githubusercontent.com/3069065/129235065-483dc148-cc64-43fe-826b-334771a95604.mp4

#### Desktop

https://user-images.githubusercontent.com/3069065/129235081-9b4dadd4-1632-4af8-ad70-a13ac33f3d7b.mp4

#### iOS

https://user-images.githubusercontent.com/3069065/129235210-eb744c66-c388-4c9b-9e2c-79ebba0d6bfa.mp4

#### Android

https://user-images.githubusercontent.com/3069065/129235156-46f18b9f-2f1e-4e4f-a5c6-9457e5edc03c.mp4

### CODE REVIEW

File: src/pages/settings/PasswordPage.js
Comment (alex-mechler) [inline src/pages/settings/PasswordPage.js:78]: Just realized this, but this check does nothing, since we are setting variables in the `else` blocks.

**Review by alex-mechler — DISMISSED** [review]

File: src/pages/settings/PasswordPage.js
Comment (mananjadhav) [reply src/pages/settings/PasswordPage.js:78]: Well, if I enter mismatch passwords first and then I update the `newPassword` to match the `confirmPassword`, the `onBlureNewPassword` still showed the `Password must match` error. Hence, I added this check.

Comment (mananjadhav) [reply src/pages/settings/PasswordPage.js:78]: Okay. Got it. Will update. I thought it was about the `else` block.

Comment (mananjadhav) [reply src/pages/settings/PasswordPage.js]: Updated

**Review by alex-mechler — APPROVED** [review]

**Comment by parasharrajat** [comment]
@mananjadhav You can use the Fixes keyword to close that PR. Fixes #4606

**Comment by mananjadhav** [comment]
@parasharrajat Okay. Is it that for fresh tasks/issues we don't use Fixes and for regression we do? I am told adding Fixes closes the issue when PR is merged.

**Comment by parasharrajat** [comment]
No, it's not like that. It is just that there is nothing left to do with that issue after you have merged the fix. So it saves some time.

**Comment by mananjadhav** [comment]
Thanks for the tip.

**Comment by mananjadhav** [comment]
@AndrewGable Can you please review this and probably merge too? #4665 is blocked by this.

**Comment by AndrewGable** [comment]
Trying to untangle the duplicates before merging this.

**Review by AndrewGable — APPROVED** [review]

**Comment by AndrewGable** [comment]
CP-ing to resolve deploy blocker.

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-08-24.  :confetti_ball:

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-09-06.  :confetti_ball:

**Pull request merged**
